functions/public/Invoke-WPFGetIso.ps1

```diff
@@ -77,17 +77,58 @@ function Invoke-WPFGetIso {
             return
         }
     } elseif ($sync["ISOdownloader"].IsChecked) {
+        # Create folder browsers for user-specified locations
+        [System.Reflection.Assembly]::LoadWithPartialName("System.Windows.Forms") | Out-Null
+        $isoDownloaderFBD = New-Object System.Windows.Forms.FolderBrowserDialog
+        $isoDownloaderFBD.Description = "Please specify the path to download the ISO file to:"
+        $isoDownloaderFBD.ShowNewFolderButton = $true
+        if ($isoDownloaderFBD.ShowDialog() -ne [System.Windows.Forms.DialogResult]::OK)
+        {
+            return
+        }
+
+        # Grab the location of the selected path
+        $targetFolder = $isoDownloaderFBD.SelectedPath
+
         # Auto download newest ISO
         # Credit: https://github.com/pbatard/Fido
         $fidopath = "$env:temp\Fido.ps1"
-        $originalLocation = Get-Location
+        $originalLocation = $PSScriptRoot
 
         Invoke-WebRequest "https://github.com/pbatard/Fido/raw/master/Fido.ps1" -OutFile $fidopath
 
         Set-Location -Path $env:temp
-        & $fidopath -Win 'Windows 11' -Rel $sync["ISORelease"].SelectedItem -Arch "x64" -Lang $sync["ISOLanguage"].SelectedItem -Ed "Windows 11 Home/Pro/Edu"
+        # Detect if the first option ("System language") has been selected and get a Fido-approved language from the current culture
+        if ($sync["ISOLanguage"].SelectedIndex -eq 0)
+        {
+            & $fidopath -Win 'Windows 11' -Rel $sync["ISORelease"].SelectedItem -Arch "x64" -Lang $(Get-FidoLangFromCulture -langName "$((Get-Culture).Name)") -Ed "Windows 11 Home/Pro/Edu"
+        }
+        else
+        {
+            & $fidopath -Win 'Windows 11' -Rel $sync["ISORelease"].SelectedItem -Arch "x64" -Lang $sync["ISOLanguage"].SelectedItem -Ed "Windows 11 Home/Pro/Edu"
+        }
```

Comment:
Quick suggestion for readability and removing duplicate code:
```suggestion
        $lang = if ($sync["ISOLanguage"].SelectedIndex -eq 0) {
            Get-FidoLangFromCulture -langName (Get-Culture).Name
        } else {
            $sync["ISOLanguage"].SelectedItem
        }

        & $fidopath -Win 'Windows 11' -Rel $sync["ISORelease"].SelectedItem -Arch "x64" -Lang $lang -Ed "Windows 11 Home/Pro/Edu"
```

Reply:
@momcilovicluka, don't know if this will work on PowerShell 5, but I'll try this anyway

Reply:
I'm glad it worked